### PR TITLE
Fix staging/src/k8s.io/apimachinery/pkg/labels golint findings

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What type of PR is this?**
/kind cleanup

**Which issue(s) this PR fixes**:

Part of #68026

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/.golint_failures [hunk 1]
```diff
@@ -261,7 +261,6 @@ staging/src/k8s.io/apimachinery/pkg/apis/meta/v1beta1
 staging/src/k8s.io/apimachinery/pkg/apis/testapigroup
 staging/src/k8s.io/apimachinery/pkg/apis/testapigroup/v1
 staging/src/k8s.io/apimachinery/pkg/conversion
-staging/src/k8s.io/apimachinery/pkg/labels
```
Comment (habibrosyad) [inline hack/.golint_failures:264]: It's look like this is for golint failures (not staticchecks), maybe needs a retitle and updating the descriptions.

Comment (ialidzhikov) [reply hack/.golint_failures:264]: Thank you! I now updated the PR title and the commit message.

File: staging/src/k8s.io/apimachinery/pkg/labels/selector.go [hunk 2]
```diff
@@ -330,51 +330,51 @@ func safeSort(in []string) []string {
 }
 
 // Add adds requirements to the selector. It copies the current selector returning a new one
-func (lsel internalSelector) Add(reqs ...Requirement) Selector {
-	var sel internalSelector
-	for ix := range lsel {
-		sel = append(sel, lsel[ix])
+func (s internalSelector) Add(reqs ...Requirement) Selector {
+	var new internalSelector
```
Comment (caesarxuchao) [inline staging/src/k8s.io/apimachinery/pkg/labels/selector.go:334]: s/new/ret

`new` is a keyword, let's avoid using it.

**Review by caesarxuchao — COMMENTED** [review]
A nit. otherwise lgtm.

/ok-to-test
/approve

**Comment by ialidzhikov** [comment]
@lavalamp could you please check the `pkg/registry/` and `staging/src/k8s.io/api/` parts?

**Pull request merged**